src/SimplePie.php

```diff
@@ -1742,25 +1745,27 @@ protected function fetch_data(&$cache)
                             $headers['if-none-match'] = $this->data['headers']['etag'];
                         }
 
-                        $file = $this->registry->create(File::class, [$this->feed_url, $this->timeout/10, 5, $headers, $this->useragent, $this->force_fsockopen, $this->curl_options]);
-                        $this->status_code = $file->status_code;
-
-                        if ($file->success) {
-                            if ($file->status_code === 304) {
-                                // Set raw_data to false here too, to signify that the cache
-                                // is still valid.
-                                $this->raw_data = false;
-                                $cache->set_data($cacheKey, $this->data, $this->cache_duration);
-                                return true;
-                            }
-                        } else {
+                        try {
+                            $file = $this->get_http_client($this->timeout / 10)->request(Client::METHOD_GET, $this->feed_url, $headers);
```

Comment:
`$this->timeout/10` could also lead to `float` instead of `int`.

Reply:
Thanks, fixed.